### ISSUE TITLE
Android firmware error

### ISSUE THREAD

**Issue by zebrum** [AdguardTeam/AdguardForAndroid#1004]
id 1464571

[logcat.txt](https://github.com/AdguardTeam/AdguardForAndroid/files/678561/logcat.txt)

```

12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   | held mutexes=
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   kernel: (couldn't read /proc/self/task/11141/stack)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   native: #00 pc 00012e02  /system/lib/libc.so (syscall+34)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   native: #01 pc 00101167  [stack:11141] (???)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.lang.Object.wait!(Native method)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   - waiting on <0x21afb84c> (a java.lang.Object)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.lang.Object.wait(Object.java:422)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.commons.b.o.a(unavailable:-1)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   - locked <0x21afb84c> (a java.lang.Object)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.commons.b.o.read(unavailable:-1)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   - locked <0x21afb84c> (a java.lang.Object)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.commons.b.q.read(unavailable:-1)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.commons.b.p.read(unavailable:-1)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.commons.b.p.read(unavailable:-1)
12-28 17:19:24.132 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.io.FilterInputStream.read(FilterInputStream.java:118)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.commons.b.n.a(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.commons.b.n.read(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   - locked <0x38c0d095> (a com.adguard.commons.b.n)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at org.apache.commons.io.input.ProxyInputStream.read(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at org.bouncycastle.util.io.Streams.readFully(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at org.bouncycastle.util.io.Streams.readFully(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at org.bouncycastle.crypto.tls.TlsUtils.readAllOrNothing(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at org.bouncycastle.crypto.tls.RecordStream.readRecord(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at org.bouncycastle.crypto.tls.TlsProtocol.safeReadRecord(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at org.bouncycastle.crypto.tls.TlsProtocol.readApplicationData(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at org.bouncycastle.crypto.tls.TlsInputStream.read(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.io.InputStream.read(InputStream.java:162)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at org.bouncycastle.crypto.tls.TlsInputStream.read(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.io.FilterInputStream.read(FilterInputStream.java:114)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.commons.b.m.d(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.filter.b.a.a(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.filter.b.a.<init>(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.filter.b.g.<init>(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.filter.proxy.a.b(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.filter.proxy.a.a(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.filter.proxy.a.a(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.android.filtering.filter.i.a(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.filter.proxy.a.a(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.android.filtering.filter.i.a(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.filter.proxy.n.a(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at com.adguard.filter.proxy.n$2.run(unavailable:-1)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.lang.Thread.run(Thread.java:818)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285] 
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285] "pool-2-thread-315" prio=5 tid=94 TimedWaiting
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | group="" sCount=0 dsCount=0 obj=0x1f5b4160 self=0x9c056600
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | sysTid=11139 nice=0 cgrp=apps sched=0/0 handle=0x9bb35000
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | state=S schedstat=( 130128001 265565394 196 ) utm=12 stm=1 core=2 HZ=100
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | stack=0x9713a000-0x9713c000 stackSize=1036KB
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | held mutexes=
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   kernel: (couldn't read /proc/self/task/11139/stack)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   native: #00 pc 00012e02  /system/lib/libc.so (syscall+34)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   native: #01 pc 00101677  [stack:11139] (???)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.lang.Object.wait!(Native method)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   - waiting on <0x11e465aa> (a java.lang.Object)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.lang.Thread.parkFor(Thread.java:1220)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   - locked <0x11e465aa> (a java.lang.Object)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at sun.misc.Unsafe.park(Unsafe.java:299)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:197)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:429)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:331)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:910)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1035)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1097)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.lang.Thread.run(Thread.java:818)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285] 
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285] "pool-2-thread-312" prio=5 tid=95 TimedWaiting
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | group="" sCount=0 dsCount=0 obj=0x1f5b3da0 self=0x9c055700
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | sysTid=11136 nice=0 cgrp=apps sched=0/0 handle=0x9bb34880
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | state=S schedstat=( 23571118 65994206 107 ) utm=2 stm=0 core=3 HZ=100
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | stack=0x97446000-0x97448000 stackSize=1036KB
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   | held mutexes=
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   kernel: (couldn't read /proc/self/task/11136/stack)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   native: #00 pc 00012e02  /system/lib/libc.so (syscall+34)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   native: #01 pc 00101677  [stack:11136] (???)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.lang.Object.wait!(Native method)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   - waiting on <0x00b5459b> (a java.lang.Object)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.lang.Thread.parkFor(Thread.java:1220)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   - locked <0x00b5459b> (a java.lang.Object)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at sun.misc.Unsafe.park(Unsafe.java:299)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:197)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:429)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:331)
12-28 17:19:24.133 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:910)
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1035)
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1097)
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   at java.lang.Thread.run(Thread.java:818)
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285] 
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285] "pool-2-thread-314" prio=5 tid=96 Sleeping
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   | group="" sCount=0 dsCount=0 obj=0x1f5b3fe0 self=0x9c056100
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   | sysTid=11138 nice=0 cgrp=apps sched=0/0 handle=0x9bb34d80
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   | state=S schedstat=( 88219519 228682422 1047 ) utm=3 stm=5 core=0 HZ=100
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   | stack=0x9723e000-0x97240000 stackSize=1036KB
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   | held mutexes=
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   kernel: (couldn't read /proc/self/task/11138/stack)
12-28 17:19:24.134 23212 10999 F art     : art/runtime/runtime.cc:285]   native: #00 pc 00012e02  /system/lib/libc.so (syscall+34)
12-28 17:19:24.134 23212 10999 F art     : art/runtim...

```

**Comment by ameshkov** [comment]
Seems to be a duplicate of #1005
